### PR TITLE
chrome-helper: Drop support for auto-installing Chrome

### PR DESCRIPTION
**Pull request by pwithnall**
Drop the `--initial-setup` argument from
`eos-google-chrome-installer.py`, and the auto-start desktop file which
launched that on first boot.

All Endless OS images now default to Chromium (T32407), so
auto-installing Chrome is no longer necessary. Until we can remove the
Chrome launcher from the icon grid by default (T32409), however, the
launcher still needs to be hooked up to a way to install Chrome, which
continues to be `eos-google-chrome-installer.py` (without the
`--initial-setup` argument).

The config file and stamp files for auto-installation are no longer
needed. The legacy stamp file for `eos-google-chrome-installer.py`, in
the user’s home directory, is unlinked directly. The installer now
unconditionally calls `eos-google-chrome-helper.py` to unlink the
system-wide stamp file.

`eos-google-chrome-helper.py` (and its polkit rules) can be removed in
one or two releases’ time, once there’s a chance they’ve been called
and removed the stamp files on existing systems. This will leave those
systems just a little bit tidier.

Signed-off-by: Philip Withnall <pwithnall@endlessos.org>

https://phabricator.endlessm.com/T32410

### CODE REVIEW

**Comment by pwithnall** [comment]
Up for review, but I have not yet tested it.

**Comment by pwithnall** [comment]
Tested and pushed some fixes so that:
 * The stamp file is actually removed when starting Chrome, rather than when installing
 * Removing the stamp file doesn’t fail if the stamp file doesn’t exist

Tested on an unlocked master image by manually copying in the changed files and running `eos-google-chrome`.

**Comment by pwithnall** [comment]
@dbnicholson, would you have time to review this?

**Comment by dbnicholson** [comment]
> @dbnicholson, would you have time to review this?

Sorry, slow day. I'll review this for tomorrow.

**Review by dbnicholson — APPROVED** [review]
Looks good. I left a minor comment that you can take or leave. Feel free to merge either way.

**Comment by dbnicholson** [comment]
Oh, one other thought. In the cover letter you say we can remove the system helper once we're assured it's been run to delete the system stamp file. I like the tidying up, but "a couple more releases" doesn't actually mean anything until a checkpoint.

An alternate fix is just to blow all it away now and add a tmpfiles snippet that deletes the initial-setup-done file:
```
r /var/lib/eos-google-chrome-helper/initial-setup-done
```

**Comment by pwithnall** [comment]
> An alternate fix is just to blow all it away now and add a tmpfiles snippet that deletes the initial-setup-done file:
> 
> ```
> r /var/lib/eos-google-chrome-helper/initial-setup-done
> ```

You are very clever and this is a good suggestion.

File: chrome-helper/eos-google-chrome.py [hunk 1]
```diff
@@ -42,13 +42,33 @@ def exit_with_error(message):
...
 
         self._start()
 
+    def _remove_done_file(self):
+        # The helper previously touched a system-wide stamp file, but since
+        # auto-installation is no longer supported, that’s no longer needed.
+        # So delete it.
+        system_helper_cmd = os.path.join(config.PKG_DATADIR, 'eos-google-chrome-system-helper.py')
+        try:
+            subprocess.check_call('pkexec {} --reset'.format(system_helper_cmd), shell=True)
```
[the remark this answers is not in the document]
Comment (pwithnall) [reply chrome-helper/eos-google-chrome.py:67]: That’s how the code was before, but improvements can certainly be made.

[the remark this answers is not in the document]
Comment (pwithnall) [reply chrome-helper/eos-google-chrome.py:67]: Dropped this code in favour of tmpfiles

**Comment by pwithnall** [comment]
Updated and ready for re-review.

**Review by dbnicholson — APPROVED** [review]
Winner! Arguably the 1st commit is unnecessary since the entire program is subsequently removed, but we can keep that fix should the other parts need to be reverted.

**Comment by pwithnall** [comment]
> Arguably the 1st commit is unnecessary since the entire program is subsequently removed, but we can keep that fix should the other parts need to be reverted.

Yeah, that was my thinking

**Comment by pwithnall** [comment]
And here’s the inevitable follow-up because I forgot to install the tmpfiles.d snippet: https://github.com/endlessm/eos-browser-tools/pull/20

**Pull request merged**